src/main/java/gift/domain/option/JpaOptionRepository.java

```diff
@@ -1,13 +1,22 @@
 package gift.domain.option;
 
 import gift.domain.product.Product;
+import jakarta.persistence.LockModeType;
 import java.util.List;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface JpaOptionRepository extends JpaRepository<Option, Long> {
 
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @Query("SELECT o FROM Option o WHERE o.id = :id")
```

Comment:
요건 굳이 `@Query` 안쓰셔도 될 것 같아요~!
메서드 네이밍으로 잠금을 표현하려는 의도 좋네요!
```suggestion
    Optional<Option> findByIdForUpdate(@Param("id") Long id)
```

Reply:
앗 왜 저렇게 했었을까요.. 수정했습니다!

Reply:
잠금이 필요한 경우에 한해 사용되는 findByIdForUpdate 쿼리메서드가 findById 기능을 자동 지원하지 않아 @Query 를 사용하여 쿼리를 작성했는데 @Query 를 쓰지 않으면 저 부분을 어떻게 해야 할까요..?!

Reply:
요렇게 사용하면 되지 않을까요!?
```java
@Lock(LockModeType.PESSIMISTIC_WRITE)
Optional<Option> findById(@Param("id") Long id);
```